### PR TITLE
DIV-6496: Adding general orders to d8 attribute in Divorce session fo…

### PR DESCRIPTION
**Pull request by jvictorcf**
…rmat

# Description

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Unit and functional tests.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/domain/model/usersession/DivorceSession.java [hunk 1]
```diff
@@ -640,4 +644,12 @@ public void setD8Documents(List<UploadedFile> d8Documents) {
             this.d8Documents = d8Documents;
         }
     }
+
+    public void addD8Document(UploadedFile d8Document) {
+        if (d8Documents == null) {
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/domain/model/usersession/DivorceSession.java:649]: it'd use either `this.` everywhere or nowhere :)

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/CCDCaseToDivorceMapper.java [hunk 2]
```diff
@@ -189,20 +187,13 @@ private String translateToBooleanString(final String value) {
...
-                    divorceSession.setD8Documents(d8Documents);
-                } else {
-                    d8Documents.addAll(serviceApplicationDocuments);
-                }
-            });
+            .ifPresent(docs -> docs.forEach(divorceSession::addD8Document));
+
+        Optional.ofNullable(divorceSession.getGeneralOrders())
+            .ifPresent(docs -> docs.forEach(divorceSession::addD8Document));
     }
```
Comment (madjava) [inline src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/CCDCaseToDivorceMapper.java:197]: Nice 👌🏾

File: src/test/java/uk/gov/hmcts/reform/divorce/caseformatterservice/functionaltest/TransformToDivorceFormatITest.java [hunk 3]
```diff
@@ -50,58 +46,17 @@ public void givenCaseDataIsNull_whenTransformToDivorceFormat_thenReturnBadReques
...
-            .andExpect(status().isOk())
-            .andReturn();
-
-        final DivorceSession actualDivorceSession =
-            ObjectMapperTestUtil.convertJsonToObject(result.getResponse().getContentAsString(),
-                DivorceSession.class);
-
-        assertThat(actualDivorceSession.getServiceApplicationDocuments(), nullValue());
-        assertThat(actualDivorceSession.getD8Documents(), hasSize(2));
-    }
```
Comment (madjava) [inline src/test/java/uk/gov/hmcts/reform/divorce/caseformatterservice/functionaltest/TransformToDivorceFormatITest.java:106]: Rest in peace 🤓

**Review by madjava — APPROVED** [review]

**Review by yarpo — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/divorce/caseformatterservice/functionaltest/TransformToDivorceFormatITest.java (hunk 3 above)
Comment (jvictorcf) [reply src/test/java/uk/gov/hmcts/reform/divorce/caseformatterservice/functionaltest/TransformToDivorceFormatITest.java:106]: It's part of the "unit" test now.

**Pull request merged**